### PR TITLE
Convert ASM -> Mixins

### PR DESCRIPTION
**Pull request by Cleptomania**
This re-writes most of the ASM Transformers to be done with mixins in order for it to be more readable, maintainable, and more broadly compatible.

The only things left as ASM are a Forge transformer that loads too early for mixins to handle, and some other mod compatibility things that are not mods included in GTNH.

This also renames the package to `com.gtnewhorizons.neid` and unifies around `neid` as the Mod ID. Previously it was half things were `notenoughIDs`(breaks lowercase convention) and half `idextender`.

**REQUIREMENTS**
In order to run in the full pack, this change needs battlegear2 to be updated to 1.3.0 otherwise battlegear's ASM will throw off NEID's mixins. This should be the only collision in the pack if you are on GTNH 2.4.1(this should be not a problem in nightlies that already have a newer version of battlegear2

### CODE REVIEW

**Comment by Caedis** [comment]
Just a heads up, make sure the project builds before pushing. Parts of `Names.java` were overdeleted.
Can probably just leave that class untouched until all are converted to mixins.

**Review by Caedis — APPROVED** [review]

**Comment by Caedis** [comment]
Need someone else to review the changes I just made

**Comment by Caedis** [comment]
@mitchej123  @Alexdoru when you have a moment

File: src/main/resources/META-INF/neid_at.cfg
Comment (glowredman) [inline src/main/resources/META-INF/neid_at.cfg:1]: Remove empty file

File: gradle.properties [hunk 1]
```diff
@@ -33,33 +33,34 @@ developmentEnvironmentUserName = Developer
...
+replaceGradleTokenInFile=
 # In case your mod provides an API for other mods to implement you may declare its package here. Otherwise, you can
 # leave this property empty.
 # Example value: apiPackage = api + modGroup = com.myname.mymodid -> com.myname.mymodid.api
 apiPackage =
 
 # Specify the configuration file for Forge's access transformers here. It must be placed into /src/main/resources/META-INF/
 # Example value: mymodid_at.cfg
-accessTransformersFile =
+accessTransformersFile = neid_at.cfg
```
Comment (glowredman) [inline gradle.properties:50]: AT conig specified is empty, please remove.

**Review by glowredman — COMMENTED** [review]
Looks good in general, I left some comments though. Please also add comments to the magic numbers used in the various classes.

File: src/main/java/com/gtnewhorizons/neid/NEIDConfig.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+public class NEIDConfig {
+
+    static Configuration config;
+    public static boolean catchUnregisteredBlocks;
+    public static boolean removeInvalidBlocks;
+    public static boolean postNeidWorldsSupport;
+    public static boolean extendDataWatcher;
+
+    public static void init(final File file) {
```
Comment (Alexdoru) [inline src/main/java/com/gtnewhorizons/neid/NEIDConfig.java:16]: why is this public static, is it used anywhere else ? can't you just make all of that a static block

File: src/main/java/com/gtnewhorizons/neid/mixins/TargetedMod.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
+package com.gtnewhorizons.neid.mixins;
+
+public enum TargetedMod {
```
Comment (Alexdoru) [inline src/main/java/com/gtnewhorizons/neid/mixins/TargetedMod.java:3]: you could delete this class since you only have VANILLA

Comment (mitchej123) [reply src/main/java/com/gtnewhorizons/neid/mixins/TargetedMod.java:3]: We'll potentially need it for the meta expansion.

File: src/main/java/com/gtnewhorizons/neid/mixins/Mixins.java [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+    ).setApplyIf(() -> NEIDConfig.extendDataWatcher));
+    // spotless:on
+    private final List<String> mixinClasses;
+    private final List<TargetedMod> targetedMods;
+    private final List<TargetedMod> excludedMods;
+    private final Supplier<Boolean> applyIf;
+    private final Phase phase;
+    private final Side side;
+
+    Mixins(Builder builder) {
```
[the remark this answers is not in the document]
Comment (mitchej123) [reply src/main/java/com/gtnewhorizons/neid/mixins/Mixins.java:49]: I really need to refactor the builder into gtnhlib or unimixins or something....

File: src/main/java/com/gtnewhorizons/neid/mixins/Mixins.java [hunk 5]
```diff
@@ -0,0 +1,209 @@
...
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+import java.util.function.Supplier;
+
+import com.gtnewhorizons.neid.NEIDConfig;
+
+import cpw.mods.fml.relauncher.FMLLaunchHandler;
+
+public enum Mixins {
```
Comment (Alexdoru) [inline src/main/java/com/gtnewhorizons/neid/mixins/Mixins.java:14]: this whole class is overkill, you can just do : 
`getMixins() {
List<String> list = new ArrayList<>();
if (FMLLaunchHandler.side().isClient()) {
list.add ()// client mixins
} else if (FMLLaunchHandler.side().isServer()) {
list.add ()// server mixins
}
list.add ()// bothj sides mixins
return list;
}`

File: src/main/java/com/gtnewhorizons/neid/mixins/early/minecraft/MixinExtendedBlockStorage.java [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+@Mixin(ExtendedBlockStorage.class)
+public class MixinExtendedBlockStorage implements IExtendedBlockStorageMixin {
+
+    @Shadow
+    private int blockRefCount;
+
+    @Shadow
+    private int tickRefCount;
+
+    // TODO: This can be made private after more of the ASM is removed.
```
Comment (Alexdoru) [inline src/main/java/com/gtnewhorizons/neid/mixins/early/minecraft/MixinExtendedBlockStorage.java:26]: why is that comment here ? the thing is already private, either the comment or the code needs to be changed

File: src/main/java/com/gtnewhorizons/neid/mixins/early/minecraft/MixinS21PacketChunkData.java [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+import com.gtnewhorizons.neid.mixins.interfaces.IExtendedBlockStorageMixin;
+
+@Mixin(S21PacketChunkData.class)
+public class MixinS21PacketChunkData {
+
+    @Shadow
+    private static byte[] field_149286_i;
+
+    @Inject(method = "<init>()V", at = @At(value = "RETURN"), require = 1)
+    private void notenoughIDs$S21PacketChunkDataConstructorAddition(CallbackInfo CI) {
```
Comment (Alexdoru) [inline src/main/java/com/gtnewhorizons/neid/mixins/early/minecraft/MixinS21PacketChunkData.java:24]: can you remove the `S21PacketChunkData` from the method names in this whole class

File: src/main/java/ru/fewizz/idextender/IEPlugin.java [hunk 8]
```diff
@@ -1,34 +1,40 @@
...
+ *
+ * @author Cleptomania
+ */
+@IFMLLoadingPlugin.MCVersion("1.7.10")
 public class IEPlugin implements IFMLLoadingPlugin {
 
+    @Override
     public String[] getASMTransformerClass() {
-        return new String[] { IETransformer.class.getName() };
+        return new String[] {};
```
Comment (Alexdoru) [inline src/main/java/ru/fewizz/idextender/IEPlugin.java:19]: you can return null

**Comment by Alexdoru** [comment]
can you change all the method handlers that start with `notenoughIDs$` to `neid$` and also remove the name of the class from certain of the method names such as `notenoughIDs$RenderGlobalplayAuxSFXConstant1`

**Review by glowredman — APPROVED** [review]

**Pull request merged**